facad1ng.py

```diff
@@ -57,61 +59,86 @@ def print_banners():
-        break
-    except ValueError as e:
-        print(e)
-
-while True:
-    custom_domain = input(f"\n{Y}Enter your custom domain {W}(ex: gmail.com): {W}")
-    try:
-        validate_custom_domain(custom_domain)
-        break
-    except ValueError as e:
-        print(e)
-
-while True:
-    phish = input(f"\n{C}Enter phishing keywords {W}(ex: free-stuff, login): {W}")
-    phish = format_phish_keywords(phish)
-    if " " not in phish:
-        break
-    else:
-        print("Phishing keywords should not contain spaces. Use '-' to separate them.")
-
-# Prepare the data for the request
-data = {
-    'url': web_url,
-    'shorturl': '',
-}
-
-# Shorten the original URL with multiple URL shorteners
-short_urls = [shortener.short(web_url) for shortener in shorteners]
-
-# Mask the URLs with custom domain and phishing keywords
-def mask_url(domain, keyword, url):
-    # Use urlparse to properly split the URL
-    parsed_url = urlparse(url)
-
-    # Reconstruct the URL with the custom domain and phishing keyword
-    return f"{parsed_url.scheme}://{domain}-{keyword}@{parsed_url.netloc}{parsed_url.path}"
-
-# Print the results
-print(f"\n{Y}Original URL:{W}", web_url, "\n")
-print(f"{G}[~] {R}Masked URL (using multiple shorteners):{W}")
-for i, short_url in enumerate(short_urls):
-    masked_url = mask_url(custom_domain, phish, short_url)
-    print(f"{G}╰➤ {C}Shortener {W} {i + 1}: {masked_url}")
+try:
+    while True:
+        web_url = input(f"{G}Enter the original link {W}(ex: https://www.ngrok.com): {W}")
+        try:
+            validate_web_url(web_url)
+            break
+        except ValueError as e:
+            print(e)
+
+    while True:
+        custom_domain = input(f"\n{Y}Enter your custom domain {W}(ex: gmail.com): {W}")
+        try:
+            validate_custom_domain(custom_domain)
+            break
+        except ValueError as e:
+            print(e)
+
+    while True:
+        phish = input(f"\n{C}Enter phishing keywords {W}(ex: free-stuff, login): {W}")
+        phish = format_phish_keywords(phish)
+        if " " not in phish:
+            break
+        else:
+            print("Phishing keywords should not contain spaces. Use '-' to separate them.")
+
+    # Prepare the data for the request
+    data = {
+        'url': web_url,
+        'shorturl': '',
+    }
+
+    # Shorten the original URL with multiple URL shorteners
+    short_urls = [shortener.short(web_url) for shortener in shorteners]
+
+    # Mask the URLs with custom domain and phishing keywords
+    def mask_url(domain, keyword, url):
+
+        encoded_url = base64.b64encode(url.encode()).decode()
+        encoded_keywords = base64.b64encode(keyword.encode()).decode()
+        # Use urlparse to properly split the URL
+        parsed_url = urlparse(encoded_url)
+
+        # Reconstruct the URL with the custom domain and phishing keyword
+        return f"{parsed_url.scheme}://{domain}-{encoded_keywords}@{parsed_url.netloc}{parsed_url.path}"
```

Comment:
URL schema doesn't display in the final output. 
![image](https://github.com/spyboy-productions/Facad1ng/assets/33115688/cd46ad8e-a21d-4bb6-bcc7-4311884ede4b)

the URL should start with `http` or `https://`

Reply:
yeah, it works in the original code, she probably missed out something. Checking